### PR TITLE
Upgrade to Ruby 3.0.5

### PR DESCRIPTION
**Pull request by Throne3d**
Relies on our Rails 6.1 upgrade (to make gem changes conflict less). Includes an update to our pg gem to remove a deprecation warning, and makes some of our version constraints more liberal.

Initially, it seems like this is just ... working? Created a post, tested out the rich text editor, things just seem to naively work!

### CODE REVIEW

**Review by Marri — CHANGES_REQUESTED** [review]
.rubocop.yml needs updating too

**Review by Marri — APPROVED** [review]
:turtle:

File: Gemfile [hunk 1]
```diff
@@ -19,18 +19,18 @@ gem 'barnes' # heroku ruby-specific metrics
 gem 'bootstrap', '~> 4.5' # pin until major version is handled
 gem 'exception_notification'
 gem 'get_process_mem'
-gem 'gon', '~> 6.4.0'
+gem 'gon', '~> 6.4'
```
Comment (teceler) [inline Gemfile:22]: I think this will match any minor version of 6.0, which may or may not be what you intended?

Comment (Throne3d) [reply Gemfile:22]: It is! They should all be compatible, and most of the other constraints allow minor upgrades (the exceptions are rails and audited, where we have some custom patches / logic depending on internals, afaik, which we don't here)

Comment (teceler) [reply Gemfile:22]: That makes sense! just wanted to make sure it was the intended behavior, that's tripped me up before.

Comment (Throne3d) [reply Gemfile:22]: Makes sense! Thanks =)

**Pull request merged**